### PR TITLE
Adds chainId to wagmi hooks

### PR DESCRIPTION
**Pull request by jackburrus**
This PR adds chainId to the following wagmi hooks:
- useContractRead
- useBlockNumber
- useBlock

### CODE REVIEW

File: apps/hyperdrive-trading/src/ui/markets/MarketStats/LiquidityStats.tsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ export function LiquidityStats({
   hyperdrive: HyperdriveConfig;
 }): ReactElement {
   const isTailwindSmallScreen = useIsTailwindSmallScreen();
-  const { data: currentBlockNumber } = useBlockNumber();
+  const { data: currentBlockNumber } = useBlockNumber({
+    chainId: hyperdrive.chainId,
```
Comment (jackburrus) [inline apps/hyperdrive-trading/src/ui/markets/MarketStats/LiquidityStats.tsx:24]: Am I thinking about this correctly? If a user navigates to a pool from the homepage, we want the liquidity stats to pull from the hyperdrive.chain, not the chain the user is connected to. A user could navigate to a Gnosis chain pool from mainnet but we want the Gnosis chainId here.

Comment (DannyDelott) [reply apps/hyperdrive-trading/src/ui/markets/MarketStats/LiquidityStats.tsx:24]: You are correct, the pool page should be tied to the `hyperdrive.chainId`

File: apps/hyperdrive-trading/src/ui/markets/YourBalanceWell/YourBalanceWell.tsx [hunk 2]
```diff
@@ -67,6 +66,7 @@ function AvailableAsset({
   token: TokenConfig;
 }) {
   const { address: account } = useAccount();
+  const chainId = useChainId();
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/markets/YourBalanceWell/YourBalanceWell.tsx:69]: Let's use the `token.chainId` for this one

File: apps/hyperdrive-trading/src/ui/token/RevokeAllowanceModalButton.tsx [hunk 3]
```diff
@@ -51,6 +52,7 @@ export function RevokeAllowanceModalButton({
     abi: erc20Abi,
     functionName: "totalSupply",
     address: token.address,
+    chainId: chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/token/RevokeAllowanceModalButton.tsx:55]: same here, token.chainId

File: apps/hyperdrive-trading/src/ui/token/hooks/useTokenAllowance.ts [hunk 4]
```diff
@@ -21,12 +21,14 @@ export function useTokenAllowance({
...
 }: UseTokenAllowanceOptions): useTokenAllowanceResult {
   const queryEnabled = !!spender && !!account && !!tokenAddress && !!enabled;
+  const chainId = useChainId();
 
   const { data, status } = useReadContract({
     query: {
       enabled,
     },
     address: tokenAddress,
+    chainId: chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/token/hooks/useTokenAllowance.ts:31]: This should be the token's chain id, not the connected chain id.

File: apps/hyperdrive-trading/src/ui/token/hooks/useTokenBalance.ts [hunk 5]
```diff
@@ -40,6 +40,7 @@ export function useTokenBalance({
 
   const { data: tokenBalance, status: tokenBalanceStatus } = useReadContract({
     address: tokenAddress,
+    chainId: chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/token/hooks/useTokenBalance.ts:43]: token chainId

File: apps/hyperdrive-trading/src/ui/hyperdrive/MaturesOnCell/MaturesOnCell.tsx [hunk 6]
```diff
@@ -15,7 +15,8 @@ export function MaturesOnCell({
 }: {
   maturity: bigint;
 }): ReactElement {
-  const { data: currentBlock } = useBlock();
+  const chainId = useChainId();
+  const { data: currentBlock } = useBlock({ chainId });
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/MaturesOnCell/MaturesOnCell.tsx:19]: Minor: This should be the hyperdrive's chain id. `useBlock` will default to the connected chainId, so this is functionally the same as what was here before.

File: apps/hyperdrive-trading/src/ui/hyperdrive/TransactionTable/TransactionsTable.tsx [hunk 7]
```diff
@@ -517,8 +517,10 @@ function AccountCell({ account }: { account: Address }) {
 }
 
 function BlockInfo({ blockNumber }: { blockNumber: bigint | undefined }) {
+  const chainId = useChainId();
   const { data: transactionBlock, status: transactionBlockStatus } = useBlock({
     blockNumber,
+    chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/TransactionTable/TransactionsTable.tsx:523]: same here, hyperdrive.chainId

File: apps/hyperdrive-trading/src/ui/hyperdrive/hooks/useLpApy.ts [hunk 8]
```diff
@@ -14,13 +14,13 @@ export function useLpApy(hyperdriveAddress: Address): {
...
 } {
   const chainId = useChainId();
-  const { data: blockNumber } = useBlockNumber();
   const { poolInfo: currentPoolInfo } = usePoolInfo({ hyperdriveAddress });
   const appConfig = useAppConfig();
   const hyperdrive = findHyperdriveConfig({
     hyperdriveAddress,
     hyperdrives: appConfig.hyperdrives,
   });
+  const { data: blockNumber } = useBlockNumber({ chainId: hyperdrive.chainId });
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/hooks/useLpApy.ts:23]: nice!

File: apps/hyperdrive-trading/src/ui/hyperdrive/longs/OpenLongsTable/StatusCell.tsx [hunk 9]
```diff
@@ -2,10 +2,11 @@ import { CheckCircleIcon } from "@heroicons/react/24/solid";
 import classNames from "classnames";
 import { ReactElement } from "react";
 import { getRemainingTimeLabel } from "src/ui/hyperdrive/getRemainingTimeLabel";
-import { useBlock } from "wagmi";
+import { useBlock, useChainId } from "wagmi";
 
 export function StatusCell({ maturity }: { maturity: bigint }): ReactElement {
-  const { data: currentBlock } = useBlock();
+  const chainId = useChainId();
+  const { data: currentBlock } = useBlock({ chainId });
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/longs/OpenLongsTable/StatusCell.tsx:9]: hyperdrive.chainId

File: apps/hyperdrive-trading/src/ui/hyperdrive/longs/hooks/usePreviewOpenLong.ts [hunk 10]
```diff
@@ -27,11 +27,12 @@ export function usePreviewOpenLong({
   asBase,
 }: UsePreviewOpenLongOptions): UsePreviewOpenLongResult {
   const readHyperdrive = useReadHyperdrive(hyperdriveAddress);
-
+  const chainId = useChainId();
   const appConfig = useAppConfig();
   const queryEnabled = amountIn !== undefined && !!readHyperdrive;
   const { data: blockNumber } = useBlockNumber({
     watch: true,
+    chainId: chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/longs/hooks/usePreviewOpenLong.ts:35]: hyperdrive.chainId

File: apps/hyperdrive-trading/src/ui/hyperdrive/lp/hooks/usePreviewAddLiquidity.ts [hunk 11]
```diff
@@ -51,6 +52,7 @@ export function usePreviewAddLiquidity({
     enabled &&
     !!readHyperdrive;
   const { data: blockNumber } = useBlockNumber({
+    chainId: chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/lp/hooks/usePreviewAddLiquidity.ts:55]: hyperdrive.chainId

File: apps/hyperdrive-trading/src/ui/hyperdrive/shorts/OpenShortsTable/ShortRateAndSizeCell.tsx [hunk 12]
```diff
@@ -16,11 +16,15 @@ export function ShortRateAndSizeCell({
...
   const appConfig = useAppConfig();
+  const chainId = useChainId();
   const baseToken = findBaseToken({
     hyperdriveAddress: hyperdrive.address,
     appConfig,
   });
-  const { data: maturityBlock } = useBlock({ blockNumber: short.maturity });
+  const { data: maturityBlock } = useBlock({
+    blockNumber: short.maturity,
+    chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/shorts/OpenShortsTable/ShortRateAndSizeCell.tsx:26]: hyperdrive.chainId

File: apps/hyperdrive-trading/src/ui/hyperdrive/shorts/hooks/usePreviewOpenShort.ts [hunk 13]
```diff
@@ -29,11 +29,12 @@ export function usePreviewOpenShort({
 }: UsePreviewOpenShortOptions): UsePreviewOpenShortResult {
   const readHyperdrive = useReadHyperdrive(hyperdriveAddress);
   const appConfig = useAppConfig();
-
+  const chainId = useChainId();
   const queryEnabled = !!readHyperdrive && !!amountOfBondsToShort;
   const { data: blockNumber } = useBlockNumber({
     watch: true,
     query: { enabled: queryEnabled },
+    chainId: chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/shorts/hooks/usePreviewOpenShort.ts:37]: hyperdrive.chainId

File: apps/hyperdrive-trading/src/ui/token/hooks/useApproveToken.tsx [hunk 14]
```diff
@@ -47,6 +47,7 @@ export function useApproveToken({
           {
             abi: erc20Abi,
             address: tokenAddress,
+            chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/token/hooks/useApproveToken.tsx:50]: token.chainId, note we'll want to update the padding logic as well to look at token.chainId

File: apps/hyperdrive-trading/src/ui/token/hooks/useMintToken.tsx [hunk 15]
```diff
@@ -40,6 +40,7 @@ export function useMintToken({
     ],
     functionName: "maxMintAmount",
     address: token.address,
+    chainId: chainId,
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/token/hooks/useMintToken.tsx:43]: token.chainId

**Review by DannyDelott — APPROVED** [review]
Bunch of tiny ones

**Pull request merged**